homeassistant/components/updater/binary_sensor.py

```diff
@@ -45,4 +50,6 @@ def extra_state_attributes(self) -> dict | None:
             data[ATTR_RELEASE_NOTES] = self.coordinator.data.release_notes
         if self.coordinator.data.newest_version:
             data[ATTR_NEWEST_VERSION] = self.coordinator.data.newest_version
+        if self.coordinator.data.current_version:
+            data[ATTR_CURRENT_VERSION] = self.coordinator.data.current_version
```

Comment:
I do not think this attribute should be added.
1) We are generally moving away from attributes.
2) This is exactly what the [`version` integration does](https://www.home-assistant.io/integrations/version/)

Reply:
Fair point.  There is definitely a some overlap between the two integrations since you can also get the latest version from both.  Out of curiosity, what would the desired end state look like for these integrations if moved away from attributes?

Reply:
IMO this integration should not exist anymore, the version integration should implement binary_sensor for this purpose and you should use that integration instead. that one has separate entities for current/latest versions. But removing this now would be a major breaking change (and since there currently is no binary variants in the version integration it's feature set is not complete).

Historically the updater integration did 2 things, provide a binary sensor and send telemetry, the telemetry part of it was split out to the analytics integration earlier this year.

As for the attr question, as many data points as possible should be separate entities instead of attributes, attributes that are secondary that only add's extra information to the state is fine.

This PR adds the current installed version as an attribute, that is identical (even uses the same logic) to what this provides:

```yaml
sensor:
  - platform: version
```

The main difference is that this will give a separate entity that is easier for the user to use in automations and in Lovelace configuration.

Reply:
Given the size of the PR - 14 added lines including tests - I think it's OK to add this attribute.
We should - in separate PRs - mark the `updater` integration as deprecated and move the binary sensor to the `version` integration though.

Reply:
> I think it's OK to add this attribute.
 
 I do not agree. Lets stop introducing attributes as much as possible... PLEASE! 🙏

Reply:
Thanks everyone.  I will close this PR and if I get ambitious I will take a look at adding the functionality provided by the 'updater' integration to the 'version' integration and try to figure out a clean migration path.